src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using static Microsoft.Azure.WebJobs.Script.EnvironmentSettingNames;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    /// <summary>
+    /// A middleware responsible for MaxRequestBodySize size configuration
+    /// </summary>
+    internal class HttpRequestBodySizeMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IEnvironment _environment;
+        private RequestDelegate _invoke;
+        private long _maxRequestBodySize;
+
+        public HttpRequestBodySizeMiddleware(RequestDelegate next, IEnvironment environment)
+        {
+            _next = next;
+            _environment = environment;
+            _invoke = InvokeBeforeSpecialization;
+        }
+
+        // for testing
+        internal RequestDelegate InnerInvoke { get => _invoke; }
+
+        public Task Invoke(HttpContext context)
+        {
+            return _invoke(context);
+        }
+
+        internal Task InvokeAfterSpecialization(HttpContext httpContext)
+        {
+            httpContext.Features.Get<IHttpMaxRequestBodySizeFeature>().MaxRequestBodySize = _maxRequestBodySize;
```

Comment:
Unlikely to hit in our known scenarios, but a good practice is not to assume the feature is present, retrieve and check for `null`.  If it is, the option for this middleware is either to throw with a clear exception (making diagnostic easy) or noop. I would vote for the former.

Reply:
Fixed in the PR